### PR TITLE
Implement the ledger/model package

### PR DESCRIPTION
**Pull request by supl**
This PR implements the `ledger/model` package (structure ContractExecutionResult and LedgerValidationResult).

### CODE REVIEW

File: ledger/model/contract_execution_result.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		myAuditorProofs      []asset.Proof = r.AuditorProofs
+		anotherAuditorProofs []asset.Proof = another.AuditorProofs
+	)
+
+	if len(myAuditorProofs) != len(anotherAuditorProofs) {
+		return false
+	}
+
+	for i := range myAuditorProofs {
+		if !myAuditorProofs[i].Equal(anotherAuditorProofs[i]) {
```
Comment (feeblefakie) [inline ledger/model/contract_execution_result.go]: I think they might not be sorted in the same order, so I think we should make it ordering-independent?
(The Java version creates a map for one side, and looks up the map to match)

**Review by feeblefakie — COMMENTED** [review]
Thank you! Left one comment.

File: ledger/model/contract_execution_result.go (hunk 1 above)
Comment (supl) [reply ledger/model/contract_execution_result.go]: @feeblefakie 
Oh, thank you for your reminder.
I didn't realize this.

I use the map to compare too in https://github.com/scalar-labs/scalardl-go-client-sdk/pull/14/commits/02660a70b4741f53bf0572c3dc24f2cfed8fe881

Please check it.
Thank you :)

**Review by thongdk8 — APPROVED** [review]
LGTM! Thank you!

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

File: ledger/model/contract_execution_result.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+type ContractExecutionResult struct {
+	Result        dl.JSONObject
+	Proofs        []asset.Proof
+	AuditorProofs []asset.Proof
+}
+
+// Equal checks if two contract execution results have the same values.
+func (r ContractExecutionResult) Equal(another ContractExecutionResult) (equal bool) {
+	var (
+		myResult      dl.JSONObject = r.Result
```
Comment (feeblefakie) [inline ledger/model/contract_execution_result.go:21]: I think `my` and `another` should be `ledger` and `auditor` for more clarity.

**Review by feeblefakie — COMMENTED** [review]
Thank you! Left another comment. PTAL!

File: ledger/model/contract_execution_result.go (hunk 2 above)
Comment (supl) [reply ledger/model/contract_execution_result.go:21]: @feeblefakie 
Sorry for confusing you.

This function is similar to 
https://github.com/scalar-labs/scalar/blob/master/common/src/main/java/com/scalar/dl/ledger/model/ContractExecutionResult.java#L79

to check if the two contract execution results are the same.

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you! (sorry for the late review)

**Pull request merged**